Add app to Dockerfile

katlol opened the pull request:
This pull request:
- Adds app to Dockerfile
- Adds web container
- Adds docker-compose

# It has been tested locally, by:
0. Run `docker pull ghcr.io/bobcamera/bobcamera` to be able to take advantage of the latest cached Docker container. Otherwise it will take a **long** time.
1.  Creating an .env file in the folder root containing:
```
BOB_RTSP_URL=rtsp://...
```
2. Running `docker compose up --build`
3. Browsing to <http://localhost:8080/2d-stream-controls> - you should see an annotated stream.

katlol commented:
fails with https://paste.ee/r/Wut1H

fcbarros commented on docker/build-boblib-app.sh:
Why are these deleted? I use them to build locally to test it.

katlol replied:
Added back as part of a Makefile

fcbarros commented on Dockerfile:
```diff
@@ -237,3 +237,22 @@ RUN mkdir -p /opt/ros2_ws/src \
    && rosdep install --from-paths src --ignore-src --rosdistro ${ROS_DISTRO} -y \
    && colcon build --allow-overriding cv_bridge
 ENV DEBIAN_FRONTEND=
+
```
Was this tested? This is only a dev container, not the full app as we do not clone the repo here.

fcbarros replied:
The ros2 app needs to inherit from this one, clone the src and then build etc. But since I am away for a while I might be wrong.

katlol replied:
read line 251

katlol commented:
@fcbarros as requested, added 
```make build-boblib-app``` or ```make build-ros2-dev``` back which have the same functionality as the bash scripts I removed for building the image, also, it pushes to this github repo now.

Micky-G commented:
I just tried this and it works fine using, video, rtsp and usb as the source for the devcontainer. Not tried the WebUI, will leave that for @danlohman to test and ensure he is happy with those changes.
